crates/bdk/src/wallet/mod.rs

```diff
@@ -497,41 +528,80 @@ impl<D> Wallet<D> {
         &mut self,
         tx: Transaction,
         position: ConfirmationTime,
-    ) -> Result<bool, chain_graph::InsertTxError<ConfirmationTime>> {
-        let changeset = self.keychain_tracker.insert_tx(tx, position)?;
-        let changed = changeset.is_empty();
-        self.persist.stage(changeset);
+        seen_at: Option<u64>,
```

Comment:
surely you can either insert a tx in a conirmed position OR with a seent_at but not both. You should specify the `seen_at` at the Unconfirmed branch of `ConfirmationTime` or something.

Reply:
Updated to have the `ConfirmationTime::Unconfirmed` variant have a `last_seen` field.